common/observers/fs/fs.go

```diff
@@ -0,0 +1,127 @@
+package fs
+
+import (
+	"bytes"
+	"github.com/skydive-project/skydive/common"
+	"github.com/skydive-project/skydive/logging"
+	"gopkg.in/fsnotify/fsnotify.v1"
+	"io/ioutil"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// WatcherCallback described the interface for callback on matched event
+type WatcherCallback = func(details Event)
+
+// Op mimics fsnotify.Op for backward compatibility with fsnotify
+type Op fsnotify.Op
+
+// Event mimics fsnotify.Event for backward compatibility with fsnotify
+type Event fsnotify.Event
+
+// Watcher is an implementation of a wrapper around fs events
+type Watcher struct {
+	sync.Mutex
+	eventHandlers map[Op]WatcherCallback
+	fileInfo      map[string]FileInfo
+	state         int64
+}
+
+// FileInfo is a structure which describes file being tracked
+type FileInfo struct {
+	Path    string
+	Content []byte
+}
+
+// set of constants which almost mimics fsnotify type of events, except ChangedContent
+const (
+	Create Op = 1 << iota
+	Write
+	Remove
+	Rename
+	Chmod
+	ChangedContent
+)
+
+// OnEvent registers handler for specific type of event, simplifies life for developer
+func (w *Watcher) OnEvent(eventType Op, cb WatcherCallback) {
+	w.Lock()
+	defer w.Unlock()
+	logging.GetLogger().Debugf("Register event handler: %d", eventType)
+	w.eventHandlers[eventType] = cb
+}
+
+// AddNewPath starts tracking changes for new path
+func (w *Watcher) AddNewPath(path string) {
+	logging.GetLogger().Debugf("track changes for the file: %s", path)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		logging.GetLogger().Errorf("Error while reading content of the file: %s, Err: %s", path, err)
+	}
+	w.Lock()
+	defer w.Unlock()
+	w.fileInfo[path] = FileInfo{
+		Path:    path,
+		Content: data,
+	}
+	if err != nil {
+		logging.GetLogger().Errorf("Error while adding path to fs watcher: %s. Err: %s", path, err)
+	}
+}
+
+func (w *Watcher) checkFileContentEvents() {
+	for {
+		if w.isRunning() {
+			return
+		}
+		w.Lock()
+		for filePath, fileInfo := range w.fileInfo {
+			data, err := ioutil.ReadFile(filePath)
```

Comment:
It seems to be inefficient to read the whole files each 5 seconds to check if something changed. I think that it is because you are reading /sys and inotify doesn't work. So I think we should rename or move this to specific place to avoid using this for regular File systems.